detectron2/modeling/proposal_generator/rpn.py

```diff
@@ -431,24 +431,25 @@ def forward(
             for x in pred_anchor_deltas
         ]
 
-        if self.training:
+        if self.training and not torch.jit.is_scripting():
             gt_labels, gt_boxes = self.label_and_sample_anchors(anchors, gt_instances)
             losses = self.losses(
                 anchors, pred_objectness_logits, gt_labels, pred_anchor_deltas, gt_boxes
             )
-            losses = {k: v * self.loss_weight.get(k, 1.0) for k, v in losses.items()}
+            for k, v in losses.items():
+                losses[k] = v * self.loss_weight.get(k, 1.0)
         else:
             losses = {}
-
+        pred_objectness_logits = [t.detach() for t in pred_objectness_logits]
+        pred_anchor_deltas = [t.detach() for t in pred_anchor_deltas]
```

Comment:
move it into `predict_proposals`? this will make the semantics of `predict_proposals` unchanged